src/parsec_service/authenticators.md

```diff
@@ -5,6 +5,23 @@
 The direct authenticator, [currently
 named](https://github.com/parallaxsecond/parsec-interface-rs/issues/22) "simple authenticator" in
 the code, directly parse the authentication field as a UTF-8 string and uses that as application
-identity. The direct authenticator is the one currently used by the Parsec service.
+identity.
+
+## Peer Credentials Authenticator
+
+The peer credentials authenticator uses peer credentials to authenticate the client. In this
+context, 'peer credentials' refers to metadata about the connection between client and server that
+contains the effective user ID (UID) and group ID (GID) of the connecting process.
+
+To use this authenticator, the application must self-declare its UID in the authentication field of
+the request as a UTF-8 string. This authenticator will then verify that the UID sourced from the
+peer credentials matches the one self-declared in the request. If they match up, authentication is
+successful and the application identity is set to the UID.
+
+Currently, the peer credentials authenticator only supports authentication via the peer credentials
+sourced from a Unix domain socket. However, the authenticator will likely support different
```

Comment:
Not sure about the phrasing here. It seems like the reverse of what we intend. Peer credential specifically requires domain socket to be the transport, because we rely on the ability of UDS to be able to query the credential in a trusted manner. I think it's misleading to say that peer credential would support other transports in the future, unless we're willing to significantly generalise our definition of what a peer credential is (in which case I don't think we can specifically talk about Unix UIDs or GIDs).

It would be more accurate to say the reverse: that a UDS transport could be used with other auth mechanisms and not only peer credential.

Reply:
That might've been my fault, as I suggested expanding what we assume - the definition of "peer credential" would indeed depend on the transport mechanism, but I don't see an issue in assuming that other IPC mechanisms (e.g. network, d-bus) could be covered with this kind of authentication if we point out that only UDS are supported for now.

Reply:
Makes sense -- removed mentions of other transports for this section.